data/ai/lua/generic_recruit_engine.lua

```diff
@@ -330,12 +331,10 @@ return {
 
             my_hp = my_hp + sum_gold_for_sides({{"allied_with", {side = wesnoth.current.side} }})*2.3
             enemy_hp = enemy_hp+sum_gold_for_sides({{"enemy_of", {side = wesnoth.current.side} }})*2.3
-            hp_ratio = my_hp/(enemy_hp + 1e-6)
-
-            return hp_ratio
+            return my_hp / enemy_hp
```

Comment:
Did you remove the `1e-6` here intentionally?  It was obviously there to prevent a division by zero, and while that is very unlikely to happen, it is not entirely impossible, I think.

Reply:
Ah, I speculated that it was to prevent truncating division and removed it on the basis that division doesn't truncate now, but if it had a different purpose, I can put it back… although, I'm not sure it's even _sensible_? It's going to produce an `hp_ratio` a million times larger than `my_hp`.